server/src/main/java/org/opensearch/cluster/coordination/PublicationTransportHandler.java

```diff
@@ -345,6 +354,19 @@ public PublicationContext newPublicationContext(
         return publicationContext;
     }
 
+    private boolean validateRemotePublicationOnAllNodes(DiscoveryNodes discoveryNodes) {
+        if (ClusterMetadataManifest.getCodecForVersion(discoveryNodes.getMinNodeVersion()) < ClusterMetadataManifest.CODEC_V0) {
```

Comment:
when will this be true?

Reply:
This shouldn't be true. I can add an assert instead